fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalJoin.java

```diff
@@ -75,21 +76,23 @@ public class LogicalJoin<LEFT_CHILD_TYPE extends Plan, RIGHT_CHILD_TYPE extends
     // Table bitmap for tables below this join
     private long bitmap = LongBitmap.newBitmap();
 
-    private JoinHint hint;
+    private DistributeHint hint;
 
     public LogicalJoin(JoinType joinType, LEFT_CHILD_TYPE leftChild, RIGHT_CHILD_TYPE rightChild) {
-        this(joinType, ExpressionUtils.EMPTY_CONDITION, ExpressionUtils.EMPTY_CONDITION, JoinHint.NONE,
+        this(joinType, ExpressionUtils.EMPTY_CONDITION, ExpressionUtils.EMPTY_CONDITION,
+                new DistributeHint("Distribute", JoinHint.NONE),
```

Comment:
use a default "Distribute" string to avoid new each time.
Use DistributeType.NONE instead of JoinHint.NONE